src/Core/Resolvers/CosmosQueryEngine.cs

```diff
@@ -80,6 +90,48 @@ public class CosmosQueryEngine : IQueryEngine
                 queryRequestOptions.PartitionKey = new PartitionKey(partitionKeyValue);
             }
 
+            JObject executeQueryResult = null;
+
+            if (runtimeConfig.CanUseCache() && runtimeConfig.Entities[structure.EntityName].IsCachingEnabled)
+            {
+                StringBuilder dataSourceKey = new(dataSourceName);
+
+                // to support caching for paginated query adding continuation token in the datasource
+                if (structure.IsPaginated)
+                {
+                    dataSourceKey.Append(":");
+                    dataSourceKey.Append(structure.IsPaginated);
+                    dataSourceKey.Append(":");
+                    dataSourceKey.Append(structure.Continuation);
+                }
+
+                DatabaseQueryMetadata queryMetadata = new(queryText: queryString, dataSource: dataSourceKey.ToString(), queryParameters: structure.Parameters);
+
+                executeQueryResult = await _cache.GetOrSetAsync<JObject>(async () => await ExecuteQueryAsync(structure, querySpec, queryRequestOptions, container, idValue, partitionKeyValue), queryMetadata, runtimeConfig.GetEntityCacheEntryTtl(entityName: structure.EntityName));
+            }
+            else
+            {
+                executeQueryResult = await ExecuteQueryAsync(structure, querySpec, queryRequestOptions, container, idValue, partitionKeyValue);
+            }
+
+            JsonDocument response = executeQueryResult != null ? JsonDocument.Parse(executeQueryResult.ToString()) : null;
```

Comment:
We are returning this in the response, so it will be disposed of similarly to how it's happening for other APIs.

https://github.com/Azure/data-api-builder/blob/99c052dbffca6d585e8fb256e12e586c13814fbf/src/Core/Services/ResolverMiddleware.cs#L170